rotkehlchen/tests/unit/test_trades.py

```diff
@@ -148,3 +149,45 @@ def test_query_trade_history_online_but_exchange_excluded(events_historian):
 
     assert latest_trades_mock.call_count == 0
     assert iterate_exchanges_mock.call_count == 0
+
+
+@pytest.mark.parametrize('db_settings', [
+    {'non_syncing_exchanges': [ExchangeLocationID(name='Binance', location=Location.BINANCE)]}])
+def test_query_trade_history_with_exchange_instance_excluded(events_historian):
+    """Test that when ignoring the locations along with names, only the ignored location with that
+    name should be ignored and not all the instances of that location."""
+
+    def get_binances(name: list[str]) -> list[ExchangeInterface]:
+        return [
+            ExchangeInterface(
+                name,
+                Location.BINANCE,
+                ApiKey(''),
+                ApiSecret(b''),
+                events_historian.db,
+            )
+            for name in name
+        ]
+
+    binances = get_binances(['Binance', 'Binance2'])
```

Comment:
instead of doing this you can use `create_test_binance` but maybe  since binance loads more logic in the initialization that is not needed for the test you can use `create_test_coinbase` modifying it to take a name as argument as `create_test_binance` does.